lib/logstash/filters/http.rb

```diff
@@ -132,9 +133,10 @@ def sprintf_object(event, obj)
   end
 
   def process_response(body, headers, event)
-    content_type, _ = headers.fetch("content-type", "").split(";")
     event.set(@target_headers, headers)
     return if @verb == 'head' # Since HEAD requests will not contain body, we need to set only header
+
+    content_type = resolve_content_type(headers)
```

Comment:
Do you think we can check the array if "application/json" is one of the items and parse the body as json?
With multiple headers, I think "application/json" could be in any order.

Reply:
Yeah, since our interest is finding `application/json`, it also makes sense to _deep_ search.
```suggestion
    if headers_has_content_type?(headers, "application/json")
```


Remove L14 and `resolve_content_type` function will be replaced with
```
  def headers_has_content_type?(headers, looking_content_type)
    # content-type might be an array or string with ; separated
    content_types = headers.fetch("content-type", "")
    return content_types.include?(looking_content_type) unless content_types.kind_of?(Array)

    content_types.each do |content_type|
      return true if content_type.include?(looking_content_type)
    end
    false
  end
```

How do you think?

Reply:
```ruby
def headers_has_json?(headers) 
  return [ headers.fetch("content-type", "") ]
              .map{|i| i}
              .flatten
              .include?("application/json")
end
```

Reply:
I forgot to split(';'), but leave it to you

Reply:
Simplified with [this commit](https://github.com/logstash-plugins/logstash-filter-http/pull/46/commits/526d39f6ec34694da04737b9266fa5d104b1de9c).